include/envoy/buffer/buffer.h

```diff
@@ -53,6 +53,14 @@ class Instance {
   virtual void commit(RawSlice* iovecs, uint64_t num_iovecs) PURE;
 
   /**
+   * Copy out a section of the buffer.
+   * @param data supplies the output buffer to fill.
+   * @param size supplies the size of the output buffer.
+   * @param start supplies the buffer index to start copying from.
+   */
+  virtual void copyOut(void* data, uint64_t size, size_t start) const PURE;
```

Comment:
Maybe order params (start, size, data), I don't know if we formalize it, but it's kind of nice to have IN, IN/OUT, OUT ordering.

Reply:
I don't really care one way or the other, but we don't do this anywhere else (even in this file), so it seems strange to do it here. But will change.